src/compute-client/src/controller.rs

```diff
@@ -217,24 +216,32 @@ where
             self.stashed_response = Some(self.replicas.recv().await);
         }
     }
+
+    /// Drop this compute instance.
+    ///
+    /// # Panics
+    /// - If the compute instance still has active replicas.
+    pub fn drop(mut self) {
+        assert!(
+            self.replicas.get_replica_ids().next().is_none(),
+            "cannot drop instances with provisioned replicas"
+        );
+        self.replicas.send(ComputeCommand::DropInstance);
+    }
 }
 
 // Public interface
```

Comment:
There are is also an "// Internal interface" section below. Some of the changes here are just moving methods from one of the sections into the other, depending on their visibility.